iceoryx_posh/include/iceoryx_posh/popo/modern_api/base_subscriber.hpp

```diff
@@ -118,11 +118,24 @@ class BaseSubscriber
     ///            later by the user
     /// @return success if the subscriber is attached otherwise an WaitSetError enum which describes
     ///            the error
-    cxx::expected<WaitSetError> attachTo(WaitSet& waitset,
+    template <uint64_t WaitSetCapacity>
+    cxx::expected<WaitSetError> attachTo(WaitSet<WaitSetCapacity>& waitset,
                                          const SubscriberEvent subscriberEvent,
                                          const uint64_t triggerId = Trigger::INVALID_TRIGGER_ID,
                                          const Trigger::Callback<Subscriber> callback = nullptr) noexcept;
 
+    /// @brief attaches a WaitSet to the subscriber
+    /// @param[in] waitset reference to the waitset to which the subscriber should be attached to
+    /// @param[in] subscriberEvent the event which should be attached
+    /// @param[in] callback callback which is attached to the trigger and which can be called
+    ///            later by the user
+    /// @return success if the subscriber is attached otherwise an WaitSetError enum which describes
+    ///            the error
+    template <uint64_t WaitSetCapacity>
+    cxx::expected<WaitSetError> attachTo(WaitSet<WaitSetCapacity>& waitset,
+                                         const SubscriberEvent subscriberEvent,
+                                         const Trigger::Callback<Subscriber> callback) noexcept;
+
     /// @brief detaches a specified event from the subscriber, if the event was not attached nothing happens
     /// @param[in] subscriberEvent the event which should be detached
     void detachEvent(const SubscriberEvent subscriberEvent) noexcept;
```

Comment:
`detachEvent` is the opposite to `attachTo`. Normally you recognize such pairs by their names like `attachStoepsel` and `detachStoepsel`.  This is not the case here and I would prefer to make this more clear. 
In the end this are methods of the subscriber and the subscriber is interested in the events (that are collected by the waitset). `attachEvent` and `detachEvent` would be better IMHO. The waitset is then a thing that is needed to have a home for the event

Reply:
Added to #341 issue. I will deal with this in the next PR.